src/rgw/rgw_rados.cc

```diff
@@ -1505,6 +1506,7 @@ int RGWZoneParams::fix_pool_names()
   user_email_pool = fix_zone_pool_name(pool_names, name, ".rgw.users.email", user_email_pool.name);
   user_swift_pool = fix_zone_pool_name(pool_names, name, ".rgw.users.swift", user_swift_pool.name);
   user_uid_pool = fix_zone_pool_name(pool_names, name, ".rgw.users.uid", user_uid_pool.name);
+  roles_pool = fix_zone_pool_name(pool_names, name, ".rgw.authorizationdata", roles_pool.name);
```

Comment:
we'll want to update these defaults (".rgw.users.keys", etc) to use the "pool:namespace" format. i'm not sure which pools we'll want to combine initially, but maybe something like:
```
".rgw.auth:users.keys"
".rgw.auth:users.email"
".rgw.auth:users.swift"
".rgw.auth:users.uid"
".rgw.auth:roles"
```
we'll then need to parse these strings, so that the ".rgw.auth" part ends up in `rgw_bucket::name`, and the part after ':', if any, ends up in `rgw_bucket::namespace`

`fix_zone_pool_name()` will also need a lot more thought:
* it should only be enforcing uniqueness of the pool name part, and ignoring the namespaces
* it can't return a single std::string to be consumed by `rgw_bucket`'s constructor - we'd need to be able to pass in both the pool name and its namespace
* if it 'fixes' a pool name (i.e. `foo.rgw.auth` -> `foo_[random].rgw.auth`) for `user_keys_pool`, it needs to use the same value of `[random]` when it 'fixes' the pool name for other entries